packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -6688,31 +6688,42 @@ abstract class RenderObjectElement extends Element {
   }
 
   void _updateParentData(ParentDataWidget<ParentData> parentDataWidget) {
-    bool applyParentData = true;
-    assert(() {
-      try {
-        if (!parentDataWidget.debugIsValidRenderObject(renderObject)) {
-          applyParentData = false;
+    if (parentDataWidget.debugIsValidRenderObject(renderObject)) {
+      parentDataWidget.applyParentData(renderObject);
+    } else {
+      // If it's not a valid render object, then don't show the ErrorWidget, or
+      // throw an exception, just report the error and move on.
+      //
+      // This prevents it from crashing in release mode and still reports the
+      // original error in debug mode.
+      assert(() {
+        try {
+          final String parentDataWidgetType = parentDataWidget.toStringShort();
+          String errorMessage = 'Incorrect use of ParentDataWidget.\n';
+
+          if (parentDataWidgetType == 'Positioned') {
+            errorMessage += 'The widget `Positioned` must be a descendent of a `Stack` widget.\n';
```

Comment:
The wording in these two messages is different? One mentions `descendant` (there is a typo in that word, too)
and the other mentions the word `child`.

Reply:
I'll update the spelling there. So with `Positioned` it can be anywhere in the `Stack` (i.e. it does not have to be a direct child). For a `Flexible` or `Expanded`, it needs to be a *direct* child, hence the differing wording.